src/main/java/it/rate/webapp/services/GoogleImageService.java

```diff
@@ -47,7 +50,35 @@ public String savePlaceImage(MultipartFile image, Long placeId) throws IOExcepti
     try {
       return driveService.files().create(fileMeta, mediaContent).execute().getId();
     } catch (IOException e) {
-      throw new InvalidApiResponseException();
+      throw new ApiServiceUnavailableException("Could not save image to the server");
+    }
+  }
+
+  public String changeInterestImage(Long interestId, MultipartFile file) {
+
+    Interest interest = interestRepository.getReferenceById(interestId);

```

Comment:
Getting interest is not safe in this method if called from somewhere else in the future.

Reply:
Getting interest in controller done.